src/compiler/checker.ts

```diff
@@ -7063,6 +7269,13 @@ namespace ts {
                 return container.flags & NodeFlags.Static ? getTypeOfSymbol(symbol) : (<InterfaceType>getDeclaredTypeOfSymbol(symbol)).thisType;
             }
 
+            if (container.parserContextFlags & ParserContextFlags.JavaScriptFile) {

```

Comment:
This `if` and the following `if` should both be in a single `if (isInJavaScriptFile(node))`.